docs/topics/clusterdefinitions.md

```diff
@@ -41,13 +43,15 @@ $ aks-engine get-versions
+| containerRuntime                 | no                        | The container runtime to use as a backend. The default is `docker`. The other options are `kata-containers`, and `containerd`. Windows support for `containerd` is **Experimental** - see [Windows ContainerD](features.md#windows-containerd) |
 | controllerManagerConfig          | no                        | Configure various runtime configuration for controller-manager. See `controllerManagerConfig` [below](#feat-controller-manager-config)                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
 | customWindowsPackageURL          | no                        | Configure custom windows Kubernetes release package URL for deployment on Windows. The format of this file is a zip file with multiple items (binaries, cni, infra container) in it. This setting will be deprecated in a future release of aks-engine where the binaries will be pulled in the format of Kubernetes releases that only contain the kubernetes binaries.                                                                                                                                                                                                                                                                                                                 |
 | WindowsNodeBinariesURL           | no                        | Windows Kubernetes Node binaries can be provided in the format of Kubernetes release (example: https://github.com/kubernetes/kubernetes/blob/master/CHANGELOG-1.11.md#node-binaries-1). This setting allows overriding the binaries for custom builds.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    |
+| WindowsContainerdURL             | no (for development only) | **Experimental** - see [Windows ContainerD](features.md#windows-containerd) |
```

Comment:
I don't think we should change this now but i'm curious if you think these belong here or under the WindowsProfile long-term?

Reply:
a bit of history - this codebase originally supported mesos and docker swarm too, so the container runtime version existed before `kubernetesConfig`. 

I do think these may should have a cluster-wide default, but with a per-agentpool override. All of them happen to be in `kubernetesConfig`

- `containerRuntime`
- runtime version (`containerdVersion` / `mobyVersion`)
- `kubeletConfig`

So getting back to an answer - I'd consider moving them from `kubernetesConfig` to an agent pool setting as a group in the future for consistency, not just a single setting